src/nwb_datajoint/spikesorting/spikesorting_curation.py

```diff
@@ -1,127 +1,171 @@
-            the id of the parent sorting (optional)
-        parent_sorting_id : str
-        param metrics : dict
-            computed metrics for sorting (optional)
-        description : str
-            text description of this sort (optional)
-        labels : dict
-            dictionary of labels for this sort (optional)
+            The key for the original SpikeSorting
+        parent_curation_id : int, optional
+            The id of the parent sorting
+        labels : dict or None, optional
+        merge_groups : dict or None, optional
+        metrics : dict or None, optional
+            Computed metrics for sorting
+        description : str, optional
+            text description of this sort
 
         Returns
         -------
-        sorting_id : str
+        curation_key : dict
 
         """
+        if parent_curation_id == -1:
+            # check to see if this sorting with a parent of -1 has already been inserted and if so, warn the user
+            inserted_curation = (Curation & sorting_key).fetch("KEY")
+            if len(inserted_curation) > 0:
+                Warning(
+                    'Sorting has already been inserted, returning key to previously'
+                    'inserted curation')
+                return inserted_curation[0]
+
         if labels is None:
             labels = {}
         if merge_groups is None:
             merge_groups = []
         if metrics is None:
             metrics = {}
 
-        # generate a unique ID
-        found = True
-        while found:
-            id = 'C_' + str(uuid.uuid4())[:8]
-            if len((Curation & {'curation_id': id}).fetch()) == 0:
-                found = False
+        # generate a unique number for this curation
+        id = (Curation & sorting_key).fetch('curation_id')
+        if len(id) > 0:
+            id.sort()
+            curation_id = id[-1] + 1
+        else:
+            curation_id = 0
+
+        # convert unit_ids in labels to integers for labels from sortingview.
+        new_labels = {int(unit_id): labels[unit_id] for unit_id in labels}
 
-        sorting_key['curation_id'] = id
+        sorting_key['curation_id'] = curation_id
         sorting_key['parent_curation_id'] = parent_curation_id
         sorting_key['description'] = description
-        sorting_key['labels'] = labels
+        sorting_key['curation_labels'] = new_labels
         sorting_key['merge_groups'] = merge_groups
-        sorting_key['metrics'] = metrics
+        sorting_key['quality_metrics'] = metrics
+        sorting_key['time_of_creation'] = int(time.time())
 
         Curation.insert1(sorting_key)
 
-        return sorting_key['curation_id']
+        # get the primary key for this curation
+        c_key = Curation.fetch("KEY")[0]
+        curation_key = {item: sorting_key[item] for item in c_key}
+
+        return curation_key
 
     @staticmethod
-    def get_recording_extractor(key):
+    def get_recording_extractor(key: dict):
```

Comment:
how about dropping `_extractor` here? I think the new version of `spikeinterface` calls the object Recording rather than a RecordingExtractor